python/dgl/dataloading/base.py

```diff
@@ -281,9 +281,11 @@ def _find_exclude_eids_with_reverse_types(g, eids, reverse_etype_map):
     }
     exclude_eids.update(
         {
-            reverse_etype_map[k]: v
-            for k, v in exclude_eids.items()
-            if k in reverse_etype_map
+            v: F.unique(F.cat((exclude_eids[k], exclude_eids[v]), dim=0))
+            if v in exclude_eids
+            else exclude_eids[k]
+            for k, v in reverse_etype_map.items()
+            if k in exclude_eids
```

Comment:
Hmm...  This comprehension looks too complicated.  I guess it's more readable to decompose it into a for loop.
```python
for k, v in reverse_etype_map.items():
    if k in exclude_eids:
        exclude_eids[v] = ...
```

cc @frozenbugs

Reply:
It make sense, change for fix.